packages/flare/python/cron_job_ingest_events.py

```diff
@@ -1,23 +1,25 @@
 import json
-import sys
 import os
+import sys
+from datetime import date, datetime
 from typing import Optional
 
 sys.path.insert(0, os.path.join(os.path.dirname(__file__), "vendor"))
 import vendor.splunklib.client as client
-
 from flare import FlareAPI
 
 APP_NAME = "flare"
 HOST = "localhost"
 SPLUNK_PORT = 8089
 REALM = APP_NAME + "_realm"
 KV_COLLECTION_NAME = "feednext"
+CRON_JOB_MAX_RUNNING_SECONDS = 50
 
 
 def main():
+    # Cron job are stacking, so you have to make sure to stop the current one before the next one starts
```

Comment:
That looks really inefficient. Can we work around this in a smarter way? How about using the kvstore with a timestamp?
- When the job start, set a timestamp to the current date.
- After each request response, re-set the timestamp to the current date.
- When the cronjob start, look in the kv service, if the timestamp is less than 30 minutes ago, just exit right away.

Reply:
Instead of checking for a specific threshold to leave a cronjob, we could just raise a flag in the collection to say that we're actually doing work. If a new job spawns and the flag is raised, exit right away. WDYT?

Reply:
That would break forever in some crash scenarios.